### ISSUE TITLE
Issue with clumping

### ISSUE THREAD

**Issue by rskl92** [MRCIEU/TwoSampleMR#39]
Hello, 

I have read in a .csv file for my exposure using the read_exposure_data() command. When I attempt to clump the SNPs, it gives me this error message. I have also added some warning messages it has given me when I read in the exposure data, in case it may be related to the clumping problem.

Clumping jMZ6DA, 110532 SNPs
Error: lexical error: invalid char in json text.
                                       <!DOCTYPE HTML PUBLIC "-//W3C//
                     (right here) ------^

After reading in exposure data.
1: In data.table::fread(filename, header = TRUE, sep = sep) :
  C function strtod() returned ERANGE for one or more fields. The first was string input '3.39999999997601e-313'. It was read using (double)strtold() as numeric value 3.3999999999760077E-313 (displayed here using %.16E); loss of accuracy likely occurred. This message is designed to tell you exactly what has been done by fread's C code, so you can search yourself online for many references about double precision accuracy and these specific C functions. You may wish to use colClasses to read the column as character instead and then coerce that column using the Rmpfr package for greater accuracy.

Thanks

**Comment by explodecomputer** [comment]
The clumping operation seems to be working ok but the API probably won’t handle that many SNPs. You may have to split it up, try not to use more than say 1000 SNPs at a time

**Comment by rskl92** [comment]
Would I split the SNPs up by chromosome for the LD to be taken accurately into account?

**Comment by explodecomputer** [comment]
Yes, if it’s not possible for you to run this using plink directly on your own servers then I would suggest 

 
Subsetting to keep only SNPs with p < 5e-8
If that is more than 1000 SNPs then doing each of those by chromosome

**Comment by rskl92** [comment]
Hello,

I am trying to read another exposure file which has 262 SNPs and it displays the same error. What are the possible reasons that this is happening?

Clumping n2RBcG, 262 SNPs
Error: lexical error: invalid char in json text.
                                       <!DOCTYPE HTML PUBLIC "-//W3C//
                     (right here) ------^


Many thanks

**Comment by rskl92** [comment]
And to add,

This is a list of GWS sig hits for which I used this command clump_data(exposure,clump_kb=10000,clump_r2=0.01, clump_p1=5e-08, clump_p2=5e-08)

When I remove clump_p1 and clump_p2, it clumps them..

**Comment by rskl92** [comment]
Hi,

If I read in a file of an exposure dataset with SNPs from different exposures and have a phenotype identifier for them, it clumps the SNPs separately for each identifier. However, it returns the following error. Does the issue that I previously had hold for this situation even if each phenotype has less than 1000 SNPs? Does it crash because cumulatively for all phenotypes it exceeds 1000 SNPs? 
Error: lexical error: invalid char in json text.
                                       <!DOCTYPE HTML PUBLIC "-//W3C//
                     (right here) ------^
